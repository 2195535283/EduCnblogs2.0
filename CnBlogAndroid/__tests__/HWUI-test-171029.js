//__tests__/HWUI-test-171029.js
import 'react-native';
import React from 'react';
import HWUI from '../Source/UI/HomeworkPage/App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
it('renders correctly', () => {
	const tree = renderer.create(
		<HWUI />
	).toJSON();
	expect(tree).toMatchSnapshot();
});

