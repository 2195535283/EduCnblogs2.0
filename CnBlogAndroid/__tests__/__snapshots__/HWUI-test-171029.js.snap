// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "row",
        "height": 60,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Image
        source={1}
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        ClassName
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#000000",
        "height": 2,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 30,
          "fontWeight": "bold",
          "height": 40,
          "marginTop": 10,
          "textAlign": "center",
          "width": 200,
        }
      }
    >
      HomeWorks
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "marginTop": 20,
          "width": 200,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "backgroundColor": "#0588fe",
              "borderRadius": 8,
              "padding": 8,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          New HomeWork
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "flex-end",
        "marginRight": 100,
        "marginTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#00bfff",
          "fontSize": 20,
          "marginRight": 10,
          "textAlign": "center",
        }
      }
    >
      NoReply
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#00bfff",
          "fontSize": 20,
          "marginRight": 10,
          "textAlign": "center",
        }
      }
    >
      Rank
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#00bfff",
          "fontSize": 20,
          "marginRight": 10,
          "textAlign": "center",
        }
      }
    >
      Essense
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#00bfff",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      All
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "flex-end",
        "marginHorizontal": 30,
        "marginTop": 10,
      }
    }
  >
    <TextInput
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 1,
          "height": 40,
          "marginRight": 20,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "backgroundColor": "white",
            "borderRadius": 8,
            "borderWidth": 1,
            "padding": 8,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Search
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "stretch",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 30,
        "marginTop": 10,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
          Object {
            "abstract": "摘要",
            "information": "信息",
            "key": 0,
            "title": "Homework",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={undefined}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Homework
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              摘要
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#000000",
                  "fontSize": 20,
                  "textAlign": "center",
                }
              }
            >
              信息
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
